### DISCUSSION TITLE
Get static props on a dynamic page

### DISCUSSION THREAD

**Discussion by hazae41** [vercel/next.js#28493] [Help]
Hello 👋 

I have a page whose props do not change over time, so I use `getStaticProps`. 

The problem is that I also use `getInitialProps` in my `_app` page to server-side render the user avatar on every page.

Thus, the static page do not get those props because it's static.

Is there a way to have static props on a server-side rendered page? So my "static" page would have the global dynamic props plus its own static props.

**Comment by jamesmosier** [comment]
Unfortunately not. Since `getInitialProps` is executed at runtime, there is no mechanism for `getStaticProps` to run that function at build time.
